internal/command/registry/args.go

```diff
@@ -0,0 +1,203 @@
+	if flag.IsSpecified(ctx, "image") {
+		if flag.IsSpecified(ctx, "machine") || flag.IsSpecified(ctx, "select") {
+			return "", fmt.Errorf("image option cannot be used with machien and select options")
+		}
+		return flag.GetString(ctx, "image"), nil
+	}
+
+	machine, err := argsGetMachine(ctx, app)
+	if err != nil {
+		return "", err
+	}
+
+	return imageRefPath(&machine.ImageRef), nil
+}
+
+// argsGetImages returns a list of images in ImgInfo format from
+// command line args or the environment, using `org`, `app`, `running`.
+func argsGetImages(ctx context.Context) ([]ImgInfo, error) {
+	if appName := flag.GetApp(ctx); appName != "" {
+		return argsGetAppImages(ctx, appName)
+	} else if orgName := flag.GetOrg(ctx); orgName != "" {
+		return argsGetOrgImages(ctx, orgName)
+	} else if appName := appconfig.NameFromContext(ctx); appName != "" {
+		return argsGetAppImages(ctx, appName)
+	}
+	return nil, fmt.Errorf("No org or application specified")
+}
+
+// argsGetOrgImages returns a list of images for an org in ImgInfo format
+// from `running`.
+func argsGetOrgImages(ctx context.Context, orgName string) ([]ImgInfo, error) {
+	client := flyutil.ClientFromContext(ctx)
+	org, err := client.GetOrganizationBySlug(ctx, orgName)
+	if err != nil {
+		return nil, err
+	}
+
+	apps, err := client.GetAppsForOrganization(ctx, org.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	var allImgs []ImgInfo
+	for n := range apps {
+		app := &apps[n]
+		imgs, err := argsGetAppImages(ctx, app.Name)
+		if err != nil {
+			return nil, fmt.Errorf("could not fetch images for %q app: %w", app.Name, err)
+		}
+		allImgs = append(allImgs, imgs...)
+	}
+	return allImgs, nil
+
+}
+
+// argsGetAppImages returns a list of images for an app in ImgInfo format
+// from `running`.
+func argsGetAppImages(ctx context.Context, appName string) ([]ImgInfo, error) {
+	apiClient := flyutil.ClientFromContext(ctx)
+	app, err := apiClient.GetAppCompact(ctx, appName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get app %q: %w", appName, err)
+	}
+
+	flapsClient, err := flapsutil.NewClientWithOptions(ctx, flaps.NewClientOpts{
+		AppCompact: app,
+		AppName:    app.Name,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create flaps client for %q: %w", appName, err)
+	}
+	org := app.Organization
+	ctx = flapsutil.NewContextWithClient(ctx, flapsClient)
+
+	machines, err := flapsClient.ListActive(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if flag.GetBool(ctx, "running") {
+		machines = lo.Filter(machines, func(machine *fly.Machine, _ int) bool {
+			return machine.State == fly.MachineStateStarted
+		})
+	}
+
+	var imgs []ImgInfo
+	for _, machine := range machines {
+		ir := machine.ImageRef
+		imgPath := fmt.Sprintf("%s/%s@%s", ir.Registry, ir.Repository, ir.Digest)
+
+		img := ImgInfo{
+			Org:   org.Name,
+			OrgID: org.ID,
+			App:   app.Name,
+			AppID: app.ID,
+			Mach:  machine.Name,
+			Path:  imgPath,
+		}
+		imgs = append(imgs, img)
```

Comment:
`ImageInfo` should be comparable, so the easiest way to dedup this would be to collect the infos in a `map[ImgInfo]struct{}` and then use `maps.Keys()` to get back to a slice.